pkg/api/handlers/compat/images.go

```diff
@@ -231,6 +232,14 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
 
 	fromImage := mergeNameAndTagOrDigest(query.FromImage, query.Tag)
 
+	// without this early check this function would return 200 but reported error via body stream soon after
+	// it's better to let caller know early via HTTP status code that request cannot be processed
+	_, err := shortnames.Resolve(runtime.SystemContext(), fromImage)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrap(err, "failed to resolve image name"))
```

Comment:
Maybe `403 Forbidden` would be better?

Reply:
That's @jwhonce question :) He's the Yoda of REST APIs.

Reply:
Thinking about it I cannot be sure that this failed due to enforcing mode. Maybe it should be internal server error after all.